vassal-app/src/main/java/VASSAL/command/AddPiece.java

```diff
@@ -55,6 +56,19 @@ protected void executeCommand() {
         HighlightLastMoved.setLastMoved(target);
         if (GlobalOptions.getInstance().centerOnOpponentsMove()
             && !Boolean.TRUE.equals(target.getProperty(Properties.INVISIBLE_TO_ME))) {
+          if (target instanceof Stack) {
+            // Do not center on an empty Stack
+            if (((Stack) target).getPieceCount() == 0) return;
+            // Do not center on a Stack of invisible pieces
+            int visibleCount = 0;
+            for (final GamePiece gp : ((Stack) target).asList()) {
+              if (!Boolean.TRUE.equals(gp.getProperty(Properties.INVISIBLE_TO_ME))) {
+                visibleCount++;
+                break;
+              }
+            }
+            if (visibleCount == 0) return;
+          }
```

Comment:
I think you could replace that whole block with this:
```
if (target instanceof Stack &&
    !((Stack) target).asList().stream().anyMatch(PieceIterator.VISIBLE)) {
  return;
}
```

Reply:
Thanks, I new there had to be an easier way.